### DISCUSSION TITLE
Scenario when pgbouncer behind master pg node fails but master pg node itself is alive

### DISCUSSION THREAD

**Discussion by udit47** [autobase-tech/autobase#289] [General]
First of all, awesome tool for pg ha deployment. Love how detailed it is. Kudos!

Coming to the point, I looked at the template config file of pgbouncer (pgbouncer.ini.j2) and found that its pointing to postgresql running on localhost. Now, If only pgbouncer on the master node fails, then HAproxy's httpchck will not be able to detect this because it is checking Patroni API which is based on etcd state where master is still same. This will lead to failed requests. 

I know this is very specific scenario but one I can't seem to rule it out. Any thoughts?

**Comment by vitabaks** [comment]
@udit47 Yes, you are right, if pgbouncer fail, they Patroni and haproxy will  not switch the node

So far, pgbouncer health checks are only in the scheme with Consul \
another possible solution to this problem would be to set up a watchdog for the pgbouncer service.

feel free to create your PR

**Comment by udit47** [comment]
Thanks for quick reply. 

> So far, pgbouncer health checks are only in the scheme with Consul

Could you please refer me to some link for this. I can't seem to find any.

> another possible solution to this problem would be to set up a watchdog for the pgbouncer service.

So this watchdog service will restart pgbouncer if it finds failed instance of pgbouncer?
